### ISSUE TITLE
UnexpectedError "can't parse cmap: \"CMap chars: Failed reading: Can't decode hex\\\" 0066\\\"\""

### ISSUE THREAD

**Issue by MaxDaten** [Yuras/pdf-toolbox#25]
Hi it's me again ;)

I bumped against a parsing error:

```
UnexpectedError "can't parse cmap: \"CMap chars: Failed reading: Can't decode hex\\\" 0066\\\"\""
```

I can send you the pdf via e-mail, if required. pdf2text.py is able to parse the complete pdf. I tried to reexport the pdf with different tools, but the error stays.

Greetings

**Comment by Yuras** [comment]
Yes, please send the file, I'll investigate. See the email in my profile.

**Comment by Yuras** [comment]
The immediate issue is fixed via 56571faa515d5eee54e7dd4640ece468d7b48a45
I'll keep the issue open to make sure it is fixed in HEAD.

Though I see more issues with the file:

In contains multiple `beginfbchar` sections, but we read only the first one. As a result some characters are not decoded. It is similar to #24 where we had multiple `beginbfrange` sections. I created #26 for this issue.

Also some pages are completely not decoded. I'm still investigating.

Thank you for reporting!

**Comment by Yuras** [comment]
> Also some pages are completely not decoded. I'm still investigating.

E.g. page 8 consists from one XObject from. Unfortunately we don't support them yet, see #27

**Comment by MaxDaten** [comment]
Okay, good to know that this is a identifiable problem. Thanks again for investigating this

Is there maybe a workaround or tool to "normalize" pdfs I'm not aware of?

**Comment by Yuras** [comment]
> Is there maybe a workaround or tool to "normalize" pdfs I'm not aware of?

You mean a workaround for #27? No, I don't think it exists, sorry.

PDF spec is huge and there are a lot of corner cases. We have everything to handle XObject forms (e.g. its format is identical to regular page content,) it is just a question of time to handle it correctly.

**Comment by Yuras** [comment]
Fixed in master via f65706ca6c39a8002bef63c25bc0dd5dd8978850
